client/extensions/woocommerce/index.js

```diff
@@ -191,38 +191,27 @@ function addStorePage( storePage, storeNavigation ) {
 
 		analytics.pageView.record( analyticsPath, analyticsPageTitle );
 
-		renderWithReduxStore(
-			React.createElement( App, appProps, component ),
-			document.getElementById( 'primary' ),
-			context.store
-		);
+		context.primary = React.createElement( App, appProps, component );
+		next();
 	} );
 }
 
 function createStoreNavigation( context, next, storePage ) {
-	renderWithReduxStore(
-		React.createElement( StoreSidebar, {
-			path: context.path,
-			page: storePage,
-		} ),
-		document.getElementById( 'secondary' ),
-		context.store
-	);
+	context.secondary = React.createElement( StoreSidebar, {
+		path: context.path,
+		page: storePage,
+	} );
 
 	next();
 }
 
 function notFoundError( context, next ) {
-	renderWithReduxStore(
-		React.createElement( EmptyContent, {
-			className: 'content-404',
-			illustration: '/calypso/images/illustrations/illustration-404.svg',
-			title: translate( 'Uh oh. Page not found.' ),
-			line: translate( "Sorry, the page you were looking for doesn't exist or has been moved." ),
-		} ),
-		document.getElementById( 'content' ),
-		context.store
-	);
+	context.content = React.createElement( EmptyContent, {
```

Comment:
Hmm, I don't think this is going to work...

Reply:
Thanks and sorry little delayed reply here; did you referred to missing `makeLayout, clientRender` in the route? I added them so that the route can now at least render out. :-) I'm not sure yet how to get this route activated, though.

Actually this `/extensions/woocommerce/*` is a bit untested ground → definitely needs a closer look as e.g. route implementation seems little bit exotic compared to rest of Calypso.

Reply:
Referring to `context.content` (vs `primary` and `secondary`). Or maybe it _does_ work, but it's going to be fragile, since AFAIR, the `#content` `<div />` is part of the `primary` element tree.

> Actually this `/extensions/woocommerce/*` is a bit untested ground → definitely needs a closer look as e.g. route implementation seems little bit exotic compared to rest of Calypso.

Indeed. I guess we'll be able to render that 404 that currently goes to `#content` to `#primary` instead, after some tweaking maybe.

Reply:
You're right. Duh, I missed it's `content` and not `layout` (or indeed `primary`/`secondary`): in themes somewhere I saw how "not found" page was rendering directly to `context.layout` so I assumed this was the same case. Thinking of it, that must've been server side rendering part.

Reply:
> in themes somewhere I saw how "not found" page was rendering directly to `context.layout` so I assumed this was the same case. Thinking of it, that must've been server side rendering part.

Yep, that sounds familiar! But maybe we can carry at least some inspiration over -- by e.g. trying attaching the element tree to `context.layout` here, too.